### DISCUSSION TITLE
Does algorithm use stemming, etc.?

### DISCUSSION THREAD

**Discussion by Zirys1** [asreview/asreview#652] [Q&A]
Does the algorithm change the words before analysing, i.e., to lower cases, use stemming etc.? Meaning, when a user provides input and selects abstracts with the word "Behavioural" as important, will the algorithm learn that "behavioural" and/or "behaviour" might be important as well?

Thanks in advance for the answer.

**Comment by Rensvandeschoot** [comment]
Dear @Zirys1 , 

Good question!

The answer is in our Nature [Machine Intelligence paper](https://www.nature.com/articles/s42256-020-00287-7): 

> By default, ASReview converts the records’s texts into a document-term matrix, terms are converted to lowercase and no stop words are removed by default (but this can be changed). As the document-term matrix is identical in each iteration of the active learning cycle, it is generated in advance of model training and stored in the (active learning) state file. Each row of the document-term matrix can easily be requested from the state-file. Records are internally identified by their row number in the input dataset. In oracle mode, the record that is selected to be classified is retrieved from the state file and the record text and other metadata (such as title and abstract) are retrieved from the original dataset (from the file or the computer’s memory). ASReview can run on your local computer, or on a (self-hosted) local or remote server. Data (all records and their labels) remain on the users’s computer. Data ownership and confidentiality are crucial and no data are processed or used in any way by third parties. This is unique by comparison with some of the existing systems, as shown in the last column of Table 1.

Hope this answered your question!
